app/services/puppet_fact_parser.rb

```diff
@@ -5,7 +5,13 @@ def operatingsystem
     orel = os_release.dup
 
     if orel.present?
-      major, minor = orel.split('.', 2)
+      version_splits = orel.split('.')
+      if os_name == 'ubuntu'
+        major = "#{version_splits[0]}.#{version_splits[1]}"
+        minor = version_splits[2] if version_splits.length == 3
+      else
+        major, minor = version_splits
+      end
       major = major.to_s.gsub(/\D/, '')
       minor = minor.to_s.gsub(/[^\d\.]/, '')
```

Comment:
I've been thinking about trusting the Facter reported major version instead. There it reports `22.04` as the major, instead of 22 as major and 04 as minor.